### PR TITLE
support multiple subscriptions per channel

### PR DESCRIPTION
**Pull request by smnzhu**
Addresses dapperlabs/flow-go#5652

### CODE REVIEW

File: network/p2p/network.go [hunk 1]
```diff
@@ -415,14 +415,24 @@ func (n *Network) queueSubmitFunc(message interface{}) {
...
-			Err(err).
-			Str("channel_id", qm.Target.String()).
-			Str("sender_id", qm.SenderID.String()).
-			Msg("failed to process message")
+	var wg sync.WaitGroup
+
+	for _, eng := range engines {
+		wg.Add(1)
+		go func(engine network.Engine) {
+			err := engine.Process(qm.SenderID, qm.Payload)
```
Comment (vishalchangrani) [inline network/p2p/network.go:423]:
```suggestion
                          defer wg.Done()
			err := engine.Process(qm.SenderID, qm.Payload)
```

**Comment by vishalchangrani** [comment]
@smnzhu please can you also add unit test which test this new support of multiple engines for the same channel.

**Comment by smnzhu** [comment]
> @smnzhu please can you also add unit test which test this new support of multiple engines for the same channel.

Yes, will definitely do this.

**Comment by smnzhu** [comment]
After today's meeting, we will no longer be proceeding with this approach and instead will go with implementing a multiplexer component.

**Pull request closed without merging**